lib/baseTest.py

```diff
@@ -135,21 +140,49 @@ def remove_test_file(self):
         if hasattr(self, "test_url_id"):
             self.target_helper.delete_target(self.test_url_id)
 
-    def load_configs(self):
+    def extract_platform_dep_settings(self, config_value):
+        if self.current_platform_name in config_value:
+            if self.current_platform_ver in config_value[self.current_platform_name]:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name][self.current_platform_ver])
+            else:
+                platform_dep_variables = copy.deepcopy(config_value[self.current_platform_name]['default'])
+            config_value.update(platform_dep_variables)
+            return platform_dep_variables
+        return {}
+
+    # This will set new configs into variables and update if the variables already exist
+    def set_configs(self, config_variable_name, config_value):
+        # only the config in the following list can be created or updated
+        acceptable_config_list = [ConfigName.EXEC, ConfigName.INDEX, ConfigName.GLOBAL, ConfigName.FIREFOX, ConfigName.ONLINE]
+        if config_variable_name not in acceptable_config_list:
+            raise Exception('Invalid configuration name {config_name}: {config_value}'
+                            .format(config_name=config_variable_name, config_value=config_value))
+
         default_platform_dep_settings_key = "platform-dep-settings"
-        config_fp_list = ['EXEC_CONFIG_FP', 'INDEX_CONFIG_FP', 'GLOBAL_CONFIG_FP', 'FIREFOX_CONFIG_FP', 'ONLINE_CONFIG_FP']
-        current_platform_name = sys.platform
+        if default_platform_dep_settings_key in config_value:
+            # Load the index config's settings under "platform-dep-settings" base on platform
+            platform_dep_variables = self.extract_platform_dep_settings(config_value[default_platform_dep_settings_key])
+            config_value.update(platform_dep_variables)
+            config_value.pop(default_platform_dep_settings_key)
+
+        if hasattr(self, config_variable_name):
+            # getattr is a way to get variable by reference and doesn't need to be set back
+            new_config_value = getattr(self, config_variable_name)
+            new_config_value.update(config_value)
+            setattr(self, config_variable_name, new_config_value)
+        else:
+            setattr(self, config_variable_name, config_value)
+
+    def load_configs(self):
+        config_fp_list = [ConfigName.EXEC_FP, ConfigName.INDEX_FP, ConfigName.GLOBAL_FP,
+                          ConfigName.FIREFOX_FP, ConfigName.ONLINE_FP]
+
         for config_env_name in config_fp_list:
             config_variable_name = "_".join(config_env_name.split("_")[:2]).lower()
```

Comment:
`"_".join("EXEC_CONFIG_FP".rsplit("_", 1)[0].lower()` we shouldn't join again XD